tests/unit/test_contrib/test_pydantic/test_integration.py

```diff
@@ -126,3 +126,18 @@ def test(
                     "key": "other_child.val.1",
                 },
             ]
+
+
+def test_pydantic_correct_emailstr_succeeds() -> None:
```

Comment:
In addition to the integration tests, we should unit test `PydanticDTOFactory.generate_field_definitions()` using a model with types that we expect to downcast as input.